Test/QuickCheck/Arbitrary.hs

```diff
@@ -998,13 +998,10 @@ inBounds fi g = fmap fi (g `suchThat` (\x -> toInteger x == toInteger (fi x)))
 -- and its maximum absolute value depends on the size parameter.
 arbitrarySizedFractional :: Fractional a => Gen a
 arbitrarySizedFractional =
-  sized $ \n ->
-    let n' = toInteger n in
-      do b <- chooseInteger (1, precision)
-         a <- chooseInteger ((-n') * b, n' * b)
-         return (fromRational (a % b))
- where
-  precision = 9999999999999 :: Integer
+  sized $ \n -> do
+    numer <- chooseInt (-n, n)
+    denom <- chooseInt (1, max 1 n)
```

Comment:
This doesn't generate `Rational` larget than `1` does it?

Reply:
Oh it does, but it's harder to see what kind of distribution we get. Can't we just limit the precision to `min 256 size` or something like that, in the original code?

Reply:
Well, sometimes you may want to generate numerators more than 256. I have such applications, for instance.

The comment does not guarantee anything about distribution beyond "The number can be positive or negative and its maximum absolute value depends on the size parameter", which still holds. Even the range of distribution remained unchanged.

This implementation is also consistent with `smallcheck` [instance for `Ratio`](http://hackage.haskell.org/package/smallcheck-1.1.5/docs/src/Test.SmallCheck.Series.html#line-575).

Reply:
With 

```diff
 arbitrarySizedFractional =
   sized $ \n ->
     let n' = toInteger n in
-      do b <- chooseInteger (1, precision)
+      do b <- chooseInteger (1, max 256 n')
          a <- chooseInteger ((-n') * b, n' * b)
          return (fromRational (a % b))
- where
-  precision = 9999999999999 :: Integer
```

one gets

```haskell
0 % 1
(-7) % 39
505 % 129
(-349) % 98
825 % 124
1599 % 242
(-1187) % 246
(-219) % 80
(-414) % 31
1293 % 97
1103 % 158
```

--

Yet, I'd against this change alone (especially to fix #295). The `Double` and `Float` ought to have own hand written instances. @cartazio's idea to have half like above and half using full `Double` / `Float` precision is good. There are plenty of numerical bugs to be discovered by IEEE rounding issues e.g. generating (only) coarse numbers will make some users unhappy as well.

Reply:
What IEEE rounding issues do you have in mind? The proposed implementation is perfectly capable to show that associativity does not hold for `Double`, for example:

```haskell
> quickCheck $ \a b c -> a + (b + c) === ((a + b) + c :: Double)
*** Failed! Falsified (after 12 tests and 6 shrinks):
-0.2
1.0
-0.1
0.7 /= 0.7000000000000001
```

Reply:
I do not mind `Arbitrary Double` to be not just `arbitrary = arbitrarySizedFractional`, but rather `oneof [arbitrarySizedFractional, somethingElse]`, but deciding on `somethingElse` is out of scope of this particular PR, which aims to improve `arbitrarySizedFractional` first of all.

Your approach with `max 256 n'` excludes (almost all) short ratios from being generated. I would disagree that this is beneficial for any `Fractional`. This is undesirable even for `Double`: e. g., your generator is unsuitable for testing [rounding half to even](https://en.wikipedia.org/wiki/Rounding#Round_half_to_even), because it almost never generates numbers of form `ddd.5`.

Reply:
Your distribution looks like

![Screenshot from 2020-05-16 20-49-59](https://user-images.githubusercontent.com/51087/82126764-65be4880-97b7-11ea-83fc-7ec818c42aae.png)

So I don't see how it would generate `ddd.5` numbers either often enough. Yet, true, we should generate such numbers.

Reply:
Err, I'm not sure what facts about half-integers can be derived from this plot (and what distribution you desire to obtain). As quoted above, `sample` generates at least some half-integers, on contrary to the existing implementation and to your suggestion.